src/posix/platform/netif.cpp

```diff
@@ -1055,6 +1055,39 @@ static const uint8_t *getIcmp6RaMessage(const uint8_t *data, ssize_t length)
     return ret;
 }
 
+/**
+ * Returns false if the address is a Required Anycast address.
+ *
+ */
+static bool IsRequiredAnycast(const ot::Ip6::Address *aAddress)
+{
+    uint8_t prefixLength      = aAddress->GetPrefix().kLength;
+    bool    isRequiredAnycast = true;
+    int     firstBytePos      = prefixLength / 8;
+
+    if (prefixLength % 8 > 0)
+    {
+        firstBytePos++;
+    }
+
+    // Check the last byte of the prefix and the remaining bits
+    if ((aAddress->GetBytes()[firstBytePos] >> (8 - (prefixLength % 8))) != 0)
```

Comment:
```suggestion
    if ((aAddress->GetBytes()[firstBytePos] >> (8 - (prefixLength % 8))) != 0)
    if ((aAddress->GetBytes()[firstBytePos] & ((1 << (prefixLength % 8)) - 1)) != 0)
```

Consider the address `2001:db8:123:456::/62`

==>

2001:db8:123:0000 0100 0101 0110

```cpp
prefixLength = 62
8 - (prefixLength % 8) = 2
aAddress->GetBytes()[firstBytePos] >> (8 - (prefixLength % 8)) = 0b0101'01
```

Seems not correct?

Reply:
Thanks！